spec/models/card_spec.rb

```diff
@@ -1,188 +1,328 @@
+        end
+
+        it 'set correct review date, interval, repeat and attempt for repeat=1 quality=5' do
+          check_translation('house')
+          expect(card.review_date.strftime('%Y-%m-%d %H:%M')).
+              to eq((Time.zone.now + 1.days).strftime('%Y-%m-%d %H:%M'))
+          expect_parameters interval: 6, repeat: 2, attempt: 1
+        end
+
+        it 'repeat=1 quality=5' do
+          check_translation('house')
+          expect_parameters efactor: 2.6, quality: 5
+        end
+
+        it 'repeat=1 quality=4' do
+          check_translation('RoR')
+          check_translation('house')
+          expect_parameters efactor: 2.18, quality: 4
+        end
+
+        it 'repeat=1 quality=3' do
+          check_translation('RoR', 2)
+          check_translation('house')
+          expect_parameters efactor: 1.5, quality: 3
+        end
+
+        it 'repeat=1 quality=3' do
+          check_translation('RoR', 3)
+          check_translation('house')
+          expect_parameters efactor: 1.3, quality: 3
+        end
+
+        it 'repeat=1-3 quality=5' do
+          check_translation('house', 3)
+          expect(card.review_date.strftime('%Y-%m-%d %H:%M')).
+              to eq((Time.zone.now + 16.days).strftime('%Y-%m-%d %H:%M'))
+          expect_parameters interval: 45, repeat: 4, attempt: 1, efactor: 2.8, quality: 5
+        end
+      end
+
+      describe 'when interval: 6, repeat: 2, efactor: 2.6, quality: 5' do
+        before(:each) do
+          card.update_attributes(interval: 6, repeat: 2, efactor: 2.6, quality: 5)
+        end
+        it 'set correct review date, interval, repeat and attempt for repeat=2 quality=5' do
+          check_translation('house')
+          expect(card.review_date.strftime('%Y-%m-%d %H:%M')).
+              to eq((Time.zone.now + 6.days).strftime('%Y-%m-%d %H:%M'))
+          expect_parameters interval: 16, repeat: 3, attempt: 1
+        end
+
+        it 'repeat=2 quality=5' do
+          check_translation('house')
+          expect_parameters efactor: 2.7, quality: 5
+        end
+
+        it 'repeat=2 quality=4' do
+          check_translation('RoR')
+          check_translation('house')
+          expect_parameters efactor: 2.28, quality: 4
+        end
+
+        it 'repeat=2 quality=3' do
+          check_translation('RoR', 2)
+          check_translation('house')
+          expect_parameters efactor: 1.6, quality: 3
+        end
+
+        it 'repeat=2 quality=3' do
+          check_translation('RoR', 3)
+          check_translation('house')
+          expect_parameters efactor: 1.3, quality: 3
+        end
+      end
+
+      describe 'when interval: 16, repeat: 3, efactor: 2.7, quality: 5' do
+        before(:each) do
+          card.update_attributes(interval: 16, repeat: 3, efactor: 2.7, quality: 5)
+        end
+
+        it 'set correct review date, interval, repeat and attempt for repeat=3 quality=5' do
```

Comment:
Line is too long. [92/80]<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.